src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -535,35 +530,43 @@ getSMPServerClient c@AgentClient {active, smpClients, msgQ} tSess@(userId, srv,
         notifySub connId cmd = atomically $ writeTBQueue (subQ c) ("", connId, APC (sAEntity @e) cmd)
 
 reconnectServer :: AgentMonad m => AgentClient -> SMPTransportSession -> m ()
-reconnectServer c tSess = newAsyncAction tryReconnectSMPClient $ reconnections c
+reconnectServer c tSess =
```

Comment:
all calls to reconnectServer should somehow be combined with adding pending queue, so that the check that there is no reconnect loop is also atomic